File: app/src/main/java/com/ynov/ynovchat/bo/Message.java
package com.ynov.ynovchat.bo;

/**
 * Created by quentin for YnovChat on 10/12/2021.
 */
public class Message {
    String content;
    User author;
    String created_at;

    public Message(String content, User author, String created_at) {
        this.content = content;
        this.author = author;
        this.created_at = created_at;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public User getAuthor() {
        return author;
    }

    public void setAuthor(User author) {
        this.author = author;
    }

    public String getCreated_at() {
        return created_at;
    }

    public void setCreated_at(String created_at) {
        this.created_at = created_at;
    }
}

File: ynovchat_flutter/lib/bo/message.dart
import 'package:ynovchat_flutter/bo/user.dart';

class Message{
  String content;
  User author;
  String created_at;
  bool isImage = false;

  Message(this.content, this.author, this.created_at);

  Message.fromJson(Map<String, dynamic> json) :
    content = json["content"],
    author = User.fromJson(json["author"]),
    created_at = json["created_at"],
    isImage = json["isImage"];


  Map<String, dynamic> toJson() => {
    "content" : content,
    "author" : author.toJson(),
    "created_at" : created_at,
    "isImage" : isImage,
  };
}